### ISSUE TITLE
Add hostname to project settings

### ISSUE THREAD

**Issue by Pezmc** [flowforge/flowforge#1344]
### Description

> Project > Settings > Hostname 

Needs validation:
 - Host names need to be valid and unique across all projects
 - Is a valid hostname

Only shown if the Custom Domains feature is enabled, which is based on the current team type.

### Epic/Story

#324

**Comment by ZJvandeWeg** [comment]
@Pezmc I'm not sure what the problem is, some views are hidden already based on license I assume? For example billing?

As I understand it, can't we check if the EE license was applied?

**Comment by hardillb** [comment]
@ZJvandeWeg I thought this was going to be a premium feature, e.g. one available when using something other than the basic team type (and hence paying more for it)

**Comment by knolleary** [comment]
They will be a premium feature - which means the configuration options should only be available on an EE licensed instance.

What we haven't resolved is what that means in FF Cloud - which is an EE licensed instance.

The pricing principles we set out in the summer suggested that the existing Team Type we have should represent the Community level of functionality. Any new features we add that are EE licensed would be exposed in FF Cloud via a new tier of Team that comes at additional cost.

As it stands, we have not defined what that tier is, what the pricing model would be etc. We did some work to introduce the TeamType concept to the platform in a previous release. But there will be more work to do when it comes to actually introducing the new higher tier - including technical, messaging, pricing etc.

So the question is:
 - do we make this feature available to all teams on FF Cloud
    - most simple option - can gate it all on whether a license is applied or not
 - do we introduce the higher tier team in this release to include this feature?
    - will require a few days of additional work to get the technical/non-technical requirements defined for adding a new tier

My main concern at this stage of the release is the additional work needed to introduce a new Team Tier - not just technical, but all the pricing/website/messaging updates it requires and we're pretty fully committed right now.
